src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets

```diff
@@ -261,6 +261,37 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <!--
     ============================================================
+    ProjectReference Targets: Include transitive project references before 
+                              ResolveProjectReferences is called
+    - _ComputeTransitiveProjectReferences
+    - IncludeTransitiveProjectReferences
+    ============================================================
+    -->
+
+  <Target Name="_ComputeTransitiveProjectReferences"
+          DependsOnTargets="RunResolvePackageDependencies"
+          Returns="_TransitiveProjectReferences">
+    <ItemGroup>
+      <ProjectDefs Include="@(PackageDefinitions->WithMetadataValue('Type', 'project'))" />
+
+      <_TransitiveProjectReferences Include="%(ProjectDefs.Path)">
+        <ResolvedPath>%(ProjectDefs.ResolvedPath)</ResolvedPath>
+      </_TransitiveProjectReferences>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="IncludeTransitiveProjectReferences"
+          Condition="'$(ResolvePackageDependenciesForBuild)' == 'true' and '$(EnableResolvePackageDependencies)' == 'true' and Exists('$(ProjectAssetsFile)')"
+          DependsOnTargets="_ComputeTransitiveProjectReferences" 
+          BeforeTargets="AssignProjectConfiguration">
+    <ItemGroup>
+      <!-- Add the references we computed -->
+      <ProjectReference Include="@(_TransitiveProjectReferences)" />
```

Comment:
Is it possible for there to be a transitive reference from the assets file to a project that is also directly referenced?  It would be good to have a test case to cover this, where B references C, and A references both B and C.